app/src/main/java/com/trianguloy/urlchecker/modules/list/PatternModule.java

```diff
@@ -225,4 +228,100 @@ public Message(String pattern) {
             this.pattern = pattern;
         }
     }
+
+    /**
+     * On Android 10 and under, optional groups may yield a "null" in the replacement output instead of an empty string.
+     * Therefore, we just copy the implementation from newer versions
+     * https://github.com/TrianguloY/UrlChecker/issues/237
+     */
+    private static String replaceAll(String text, Pattern pattern, String replacement) {
+        // Copied from https://android.googlesource.com/platform/libcore/+/refs/heads/android13-release/ojluni/src/main/java/java/util/regex/Matcher.java#837
+        Matcher matcher = pattern.matcher(text);
+        boolean result = matcher.find();
+        if (result) {
+            StringBuffer sb = new StringBuffer();
+            int appendPos = 0;
+            do {
+                appendPos = appendReplacement(text, matcher, sb, replacement, appendPos);
+                result = matcher.find();
+            } while (result);
+            appendTail(text, sb, appendPos);
+            return sb.toString();
+        }
+        return text;
+    }
+
+    private static int appendReplacement(String text, Matcher matcher, StringBuffer sb, String replacement, int appendPos) {
+        // Copied from https://android.googlesource.com/platform/libcore/+/refs/heads/android13-release/ojluni/src/main/java/java/util/regex/Matcher.java#714
+        sb.append(text.substring(appendPos, matcher.start()));
+        appendEvaluated(matcher, sb, replacement);
+        int newAppendPos = matcher.end();
+
+        return newAppendPos;
+    }
+
+    @TargetApi(Build.VERSION_CODES.O)
```

Comment:
Is this necessary? 🤔

Reply:
`lintAlpha` gave me an error for not doing so